src/sentry/models/file.py

```diff
@@ -435,7 +435,13 @@ def delete(self, *args, **kwargs):
         blob_ids = [blob.id for blob in self.blobs.all()]
         super(File, self).delete(*args, **kwargs)
 
-        transaction.on_commit(delete_unreferenced_blobs.s(blob_ids).delay)
+        # Wait an hour to delete blobs. This helps prevent
+        # races around frequently used blobs in debug images and release files.
+        transaction.on_commit(
+            lambda: delete_unreferenced_blobs.apply_async(
+                kwargs={"blob_ids": blob_ids}, countdown=60 * 60
```

Comment:
Do you have a general idea of how many deletions we generally have pending? One thing about delaying celery tasks is that it's super inefficient about handling them - it doesn't use a priority queue, it just iterates through the list every time, from what I remember. I think this is probably fine for an hour, but if there's a huge number of tasks it could cause issues.

Reply:
After talking with ops, going down to 5 minutes and ramp up if necessary.